app/src/controllers/userController.ts

```diff
@@ -0,0 +1,127 @@
+import mongoose from 'mongoose';
+import { UserSchema } from '../models/userModel';
+import { Request, Response } from 'express';
+
+const User = mongoose.model('User', UserSchema);
+
+/**
+ * CRUD-Controller for user models
+ */
+export class UserController {
+
+    /**
+     * lists all available users
+     * @param  req express request object
+     * @param  res express response object
+     * @return     json object with statusCode and list of users
+     */
+    public async index (req: Request, res: Response) {
+        try {
+            const users = await User.find();
+            res.json({
+                statusCode: 200,
+                data: users
+            });
+        } catch(err) {
+            res.json({
+                statusCode: 400,
+                errorMessage: err.message || err.toString()
+            });
+        }
+    }
+
+    /**
+     * creates a new user instance
+     * @param  req express request object with updated user details in body
+     * @param  res express response object
+     * @return     json object with statusCode and and created user
+     */
+    public async create (req: Request, res: Response) {
+        try {
+            let newUser = new User({
+                name: req.body.name,
```

Comment:
technically it is not correct to save these fields because we are not sure for this fields. They could contain XSS, SQL or anything else. But for now it is okay. Later we'll add special middleware that filters all fields

Reply:
yes don't think it should be validated in the controller. middleware sounds good